pool/pool_machine.c

```diff
@@ -93,8 +98,34 @@ static int poolmach_state_update(struct m0_poolmach_state *st,
 	M0_ENTRY(FID_F, FID_P(&objv_real->co_id));
 
 	if (m0_conf_obj_type(objv_real) == &M0_CONF_ENCLOSURE_TYPE) {
-		st->pst_nodes_array[*idx_nodes].pn_id = objv_real->co_id;
-		M0_CNT_INC(*idx_nodes);
+		struct m0_conf_enclosure *e;
+		struct m0_conf_node      *n;
+		struct m0_poolmach_event  pme;
+		struct m0_poolnode       *pnode;
+
+		if(*idx_devices != 0)
+			M0_CNT_INC(*idx_nodes);
+
+		pnode =	&st->pst_nodes_array[*idx_nodes];
+
+		e = M0_CONF_CAST(objv_real, m0_conf_enclosure);
+		n = e->ce_node;
+		pnode->pn_id = n->cn_obj.co_id;
+		pnode->pn_index = *idx_nodes;
+		M0_LOG(M0_ALWAYS, "[ABHI] Node FID:"FID_F" Node Index:%"PRIu32" Node object:%p", FID_P(&objv_real->co_id), *idx_nodes, pnode);
+		
+		m0_conf_obj_get_lock(&n->cn_obj);
+		m0_poolnode_clink_add(poolnode_clink(pnode),
+				     &n->cn_obj.co_ha_chan);
+
+		pme.pe_type = M0_POOL_NODE;
+		pme.pe_index = *idx_nodes;
+		pme.pe_state = m0_ha2pm_state_map(n->cn_obj.co_ha_state);
+
+		M0_LOG(M0_ALWAYS, "node:"FID_F"index:%d state:%d",
```

Comment:
Change the log message from ALWAYS to DEBUG.

Reply:
Done